storage/remote/queue_manager.go

```diff
@@ -455,11 +451,12 @@ func (t *QueueManager) calculateDesiredShards() {
 
 	var (
 		timePerSample = samplesOutDuration / samplesOut
-		desiredShards = (timePerSample * (samplesIn + samplesPending + t.integralAccumulator)) / float64(time.Second)
+		desiredShards = (timePerSample * (samplesIn - samplesDropped + samplesPending + t.integralAccumulator)) / float64(time.Second)
```

Comment:
@tomwilkie can you help us out here? the desired shards calculation is still a bit confusing.
@gouthamve is thinking that even if we remove samples pending, we would still be able to upshard even if `samplesDropped` is 0 because of the integral accumulator.

Reply:
Lets look at these calculations in another PR; its important we get this one in.